### PR TITLE
Refactor: Simplify, Clarify, and More...

### PR DESCRIPTION
**Pull request by erikthedeveloper**
### Drop the idea of zero-dependencies.

Just use `moment`, `lodash`, etc... as needed! Focus on illustrating React concepts.
- [x] Pull in `lodash`. Kill `utils.js`
- [x] Pull in `classnames`. Use as-needed
- [x] Update README/repo

### REFACTOR AND SIMPLIFY

_Make sure to refactor out any questionable practices..._

- [x] Get rid of `this.DayComponent` approach
  - 📝 replaced with `EnhanceDay` render callback
- [x] Ensure various approaches/implementations are consistent.

### Annotate with comments for learning purposes

Comment for learning purposes.

### CODE REVIEW

File: src/components/Calendar/indicators.js [hunk 1]
```diff
@@ -1,38 +1,59 @@
...
+     * @param {Component} DayComponent
+     * @return {Component} The enhanced DayComponent
+     */
+    enhanceDayComponent(DayComponent) {
       const hasIndicator = (date) => this.props.events.some(
         (event) => isSameDay(event.date, date)
       );
 
-      // The enhanced props.DayComponent
-      this.DayComponent = compose([
```
Comment (erikthedeveloper) [inline src/components/Calendar/indicators.js:23]: The main idea [in this commit](https://github.com/erikthedeveloper/react-calendar-components/pull/2/commits/7643bba6db52996ed65c1e64f7fb7cc3c5637839?diff=split) is to avoid creating and setting this on `this.DayComponent`. 

While this works for this use case and doesn't break anything, this is a relatively terrible habit. This approach falls apart if `props.DayComponent` changes (more specifically if `props.*` changes, anything we derived from the previous props is now stale since we naively derived it only once in the initial `constructor`). 

Summary: Avoid teaching bad habits and focus more on the approach rather than this specific use "building this example Calendar" use case.

See [my other comment on this same commit](https://github.com/erikthedeveloper/react-calendar-components/pull/2/commits/7643bba6db52996ed65c1e64f7fb7cc3c5637839#r96118043) for more on "why I used `state`?"

File: src/components/Calendar/indicators.js [hunk 2]
```diff
@@ -1,38 +1,59 @@
...
+    }
+
+    componentDidUpdate(prevProps) {
+      // We don't want to create this on every render, but we also want
+      // to account for the fact that props.DayComponent could change
+      // We only want to re-create if props.DayComponent changed.
+      // Treat this as a "computed property".
+      if (prevProps.DayComponent !== this.props.DayComponent) {
+        const enhancedDayComponent = this.enhanceDayComponent(this.props.DayComponent);
+        this.setState({enhancedDayComponent});
```
Comment (erikthedeveloper) [inline src/components/Calendar/indicators.js]: Why compute and set this on state rather than compute in render and make use of `*ShouldUpdate`?

I feel like this is a more straightforward approach for "deriving/computing" from `props` in many cases since in order to properly go the `*ShouldUpdate` route, I would have to get into checking any other props (in this case `events: Array`) and that seems much more error-prone and case-by-case to me.

To summarize what I mean by this "computed properties as state" approach.

- Initialize `this.state = {fromX: computePropertyFromX(props.x)}`
- Watch for `props.x` changing
- Re-compute `this.setState({fromX: computePropertyFromX(newProps.x)})`

File: src/components/Calendar/indicators.js [hunk 3]
```diff
@@ -1,35 +1,52 @@
...
+
+        // 01 - Enhance the DayComponent to enable indicators
         indicatorDay,
-      ])(this.props.DayComponent);
+      ])(DayComponent);
     }
 
     render() {
-      return <Component {...this.props} DayComponent={this.DayComponent} />
+      const enhancedDayComponent = this.enhanceDayComponent(this.props.DayComponent);
```
Comment (erikthedeveloper) [inline src/components/Calendar/indicators.js]: 📝 this is not a great idea. See https://twitter.com/dan_abramov/status/820346202424246272

> It’s pretty slow to create new components during rendering. Are you 200% sure that’s the only way?

I am sure there is a better way! 🤔...

**Comment by erikthedeveloper** [comment]
To summarize the flow as-is:

1. Enhance `Calendar` with desired features via applying the related higher order components [see this usage](https://github.com/erikthedeveloper/react-calendar-components/blob/5e60482bb8fc546998953fe8c6a56c4d8177e76a/src/stories/Calendar-stories.js#L47-L50)
2. Each [`Calendar` higher order component](https://github.com/erikthedeveloper/react-calendar-components/tree/5e60482bb8fc546998953fe8c6a56c4d8177e76a/src/components/Calendar) is intended to be composable. This is primarily enabled through treating `Day` as a [Component as prop](https://github.com/erikthedeveloper/react-calendar-components/blob/5e60482bb8fc546998953fe8c6a56c4d8177e76a/src/components/Calendar/Calendar.js#L76) 
3. ...and having each `Calendar` HoC "enhance" its own `props.DayComponent` _(also via composing higher order components)_ [see this usage](https://github.com/erikthedeveloper/react-calendar-components/blob/5e60482bb8fc546998953fe8c6a56c4d8177e76a/src/components/Calendar/indicators.js#L18-L23)
4. The [`EnhanceDay` render callback component](https://github.com/erikthedeveloper/react-calendar-components/blob/5e60482bb8fc546998953fe8c6a56c4d8177e76a/src/components/Calendar/EnhanceDay.js#L4-L8) ensures that each Calendar's `props.DayComponent` is only enhanced as-needed. More generally this enables deriving/computing properties from `props` while avoiding the computed values becoming stale.

📝 See: 

- live Storybook here: https://erikthedeveloper.github.io/react-calendar-components/
- https://github.com/erikthedeveloper/react-calendar-components/tree/refactor/simplify-and-clarify/src/components
- https://github.com/erikthedeveloper/react-calendar-components/tree/refactor/simplify-and-clarify/src/stories/Calendar-stories.js
